cpp/src/gandiva/precompiled/time.cc

```diff
@@ -828,4 +841,23 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+// Convert the seconds since epoch argument to timestamp
+#define TO_TIMESTAMP(TYPE)                                      \
+  FORCE_INLINE                                                  \
+  gdv_timestamp to_timestamp##_##TYPE(gdv_##TYPE seconds) {     \
+    return static_cast<gdv_timestamp>(seconds) * MILLIS_IN_SEC; \
```

Comment:
it should be static_cast<gdv_timestamp>(seconds * MILLIS_IN_SEC) to take care of double/float values
Same for to_time
Can you add some testcases with floats/double with fractional part

Reply:
Corrected and added test cases